lib/exabgp/bgp/message/update/attribute/mpurnlri.py

```diff
@@ -41,29 +41,26 @@ def __eq__ (self, other):
 	def __ne__ (self, other):
 		return not self.__eq__(other)
 
-	def packed_attributes (self, negotiated):
+	def packed_attributes (self, negotiated, maximum):
 		if not self.nlris:
 			return
 
 		# we changed the API to nrli.pack from addpath to negotiated but not pack itself
-		maximum = negotiated.FREE_SIZE
 
-		mpurnlri = {}
+		mpurnlri = []
 		for nlri in self.nlris:
-			mpurnlri.setdefault((nlri.afi.pack(),nlri.safi.pack()),[]).append(nlri.pack(negotiated))
-
-		for (pafi,psafi),nlris in mpurnlri.iteritems():
-			payload = pafi + psafi + ''.join(nlris)
+			if nlri.family() != self.family(): # nlri is not part of specified family
+				continue
+			mpurnlri.append(nlri.pack(negotiated))
 
-			if self._len(payload) <= maximum:
+		payload = ''.join([self.afi.pack(), self.safi.pack()])
+		for nlri in mpurnlri:
```

Comment:
Yeah not sure this can ever happen either, but added a few extra checks in the code to raise a Notify exception if it does.